### PR TITLE
Bug fixes and more shared code moved to UI base

### LINKED ISSUE #574: Classic. liberate new book from top menu
In Classic. Reproduce:

* permanently delete a book from libation (remove from library, trash bin, permanently remove from libation)
* scan
* using top menu: liberate all \*
* result: stop light still red. tags and details : not downloaded
* attempt to download again will re-download. ie: it's not just a display problem; Libation doesn't recognize that this was just downloaded
* restart libation: stop light is green, and 'tags and details' = downloaded

Note, if \* step occurs by clicking stoplight instead of using top menu: no error

### PR DESCRIPTION
**Pull request by Mbucari**
## [Move filter query and RowComparer into UI base](https://github.com/rmcrackan/Libation/commit/666b5d83dfaf0dfa300945e9fc303ecb2a3ee401)

Chardonnay uses a custom comparer for sorting all grid entries (It has to be this way to work with Avalonia's DataGridColumn sorting), but it's darned useful because it takes care of sorting all episodes beneath their parents.

I also moved chardonnay's filter query into UI base.

## [Update GridEntryBindingList to behave move like Chardonnay](https://github.com/rmcrackan/Libation/commit/6800986f258ebbafeac54bd58b2e26233b45fa8f)

Chardonnay was just plain better at sorting/filtering than classic, so I made classic more like chardonnay.

## [Fix #574 (for realsies this time)](https://github.com/rmcrackan/Libation/commit/8a1b375f0dc307e9b7a3dd4d8720b6ab7a8b51db)
Pretty self explanatory. Ensure that the GridEntry's LibraryBook is always fresh.

## [Change episode default sorting to SeriesOrder descending](https://github.com/rmcrackan/Libation/commit/af8e1cd5eff9794f34888177f0f3550818ffdbca)

Episodes are sorted by series index, descending by default.

### CODE REVIEW

**Comment by Mbucari** [comment]
We need a UI test checklist to run through after making changes to the UI code.

**Comment by rmcrackan** [comment]
That would be nice. UI regression testing is a beast. We've built a sophisticated tool at work and even that's only for websites.

**Comment by Mbucari** [comment]
OK, ready for review. I thought I did something that degraded performance when adding a ton of new books on library update, but it was already like that.

File: Source/LibationUiBase/GridView/GridEntry[TStatus].cs [hunk 1]
```diff
@@ -154,9 +154,13 @@ private void UserDefinedItem_ItemChanged(object sender, string itemName)
 			if (udi.Book.AudibleProductId != Book.AudibleProductId)
 				return;
 
-			//If UserDefinedItem was changed on a different Book instance (such as when batch liberating via menus),
-			//EntryStatu's Book instance will not have the current DB state.
-			Liberate.Book = udi.Book;
+			if (udi.Book != LibraryBook.Book)
```
Comment (rmcrackan) [inline Source/LibationUiBase/GridView/GridEntry[TStatus].cs:157]: Did we not correctly update the library book elsewhere? It feels very weird for it to be a contingent UI responsibility like this.

**Comment by rmcrackan** [comment]
> I thought I did something that degraded performance when adding a ton of new books on library update, but it was already like that.

Wow, subtle burn lol

> Move filter query and RowComparer into UI base

This warms my heart. It's exactly the way I was hoping UiBase would evolve. I hope you also see it as a good thing.

File: Source/LibationUiBase/GridView/GridEntry[TStatus].cs (hunk 1 above)
Comment (Mbucari) [reply Source/LibationUiBase/GridView/GridEntry[TStatus].cs:157]: It's not really about correctly/incorrectly updating LibraryBook, and this isn't the UI.  Think MVVM. GridEntry is the ViewModel. It's been notified (via UserDefinedItem.ItemChanged) that the model has changed and is updating itself.  It then notifies the view (the actual UI) that it needs to change.

I could get into why this wasn't necessary prior to my refactor of GridEntry into LibationUiBase if you'd like, but the short version is that it was **not** better. It still had the same bug, but it was covered up by a sloppy way in which the view was updated.

**Comment by Mbucari** [comment]
>  I hope you also see it as a good thing.

Most definitely! In this case it's not really fewer lines of code, but having the bulk of the logic in one place makes it _way_ easier to maintain.

**Comment by Mbucari** [comment]
> > I thought I did something that degraded performance when adding a ton of new books on library update, but it was already like that.
> 
> Wow, subtle burn lol

If it was, it was a self-own lol. I should look into it at some point. :kicks-can:

**Comment by rmcrackan** [comment]
> Think MVVM

Fair enuf. My brain apparently naturally gravitates toward MVC where if biz logic touches UI logic then something's wrong.

Ready for merge?

**Comment by Mbucari** [comment]
> Ready for merge?

yessir. After you do, would you please mess around with the UI and see if everything is working? I've tested so many permutations that I've lost track and I can no longer break it, but I'd love a second pair of eyes. I need you to [ask it where the bathroom is](https://twitter.com/brenankeller/status/1068615953989087232?lang=en)

**Comment by rmcrackan** [comment]
https://www.youtube.com/watch?v=baY3SaIhfl0

**Comment by Mbucari** [comment]
> https://www.youtube.com/watch?v=baY3SaIhfl0

A perennial favorite, with a [heart-warming sequel](https://www.youtube.com/watch?v=ph9HGYkAiWw).

**Comment by rmcrackan** [comment]
Aww. She looks happy

Filter and newly downloaded are lookin good!

**Pull request merged**